src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java

```diff
@@ -42,98 +42,112 @@
 	 * entity instance completely.
 	 *
 	 * @param entity must not be {@literal null}.
-	 * @return the saved entity.
+	 * @return {@link Single} emitting the saved entity.
+	 * @throws IllegalArgumentException in case the given {@code entity} is {@literal null}.
 	 */
 	<S extends T> Single<S> save(S entity);
 
 	/**
 	 * Saves all given entities.
 	 *
 	 * @param entities must not be {@literal null}.
-	 * @return the saved entities.
-	 * @throws IllegalArgumentException in case the given entity is {@literal null}.
+	 * @return {@link Flowable} emitting the saved entities.
+	 * @throws IllegalArgumentException in case the given {@link Iterable entities} or one of its entities is
+	 *           {@literal null}.
 	 */
 	<S extends T> Flowable<S> saveAll(Iterable<S> entities);
 
 	/**
 	 * Saves all given entities.
 	 *
 	 * @param entityStream must not be {@literal null}.
-	 * @return the saved entities.
-	 * @throws IllegalArgumentException in case the given {@code Publisher} is {@literal null}.
+	 * @return {@link Flowable} emitting the saved entities.
+	 * @throws IllegalArgumentException in case the given {@link Flowable entityStream} is {@literal null}.
 	 */
 	<S extends T> Flowable<S> saveAll(Flowable<S> entityStream);
 
 	/**
 	 * Retrieves an entity by its id.
 	 *
 	 * @param id must not be {@literal null}.
-	 * @return the entity with the given id or {@link Maybe#empty()} if none found.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @return {@link Maybe} emitting the entity with the given id or {@link Maybe#empty()} if none found.
+	 * @throws IllegalArgumentException in case the given {@code id} is {@literal null}.
 	 */
 	Maybe<T> findById(ID id);
 
 	/**
 	 * Retrieves an entity by its id supplied by a {@link Single}.
 	 *
-	 * @param id must not be {@literal null}.
-	 * @return the entity with the given id or {@link Maybe#empty()} if none found.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @param id must not be {@literal null}. Uses the first emitted element to perform the find-query.
+	 * @return {@link Maybe} emitting the entity with the given id or {@link Maybe#empty()} if none found.
+	 * @throws IllegalArgumentException in case the given {@link Single id} is {@literal null}.
 	 */
 	Maybe<T> findById(Single<ID> id);
 
 	/**
-	 * Returns whether an entity with the given id exists.
+	 * Returns whether an entity with the given {@code id} exists.
 	 *
 	 * @param id must not be {@literal null}.
-	 * @return {@literal true} if an entity with the given id exists, {@literal false} otherwise.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @return {@link Single} emitting {@literal true} if an entity with the given id exists, {@literal false} otherwise.
```

Comment:
Ok it now gets really nitpicky but shouldn't it be `TRUE` since it is the constant from the wrapper type, not the primitive?

If you agree there are more places like this.